### ISSUE TITLE
[BUG/SUGGESTION] Multiple sessions started on system startup

### ISSUE THREAD

**Issue by brgstrm** [SamSchott/maestral#188]
**Describe the bug**
If the "Start Maestral on login" is checked in the GUI, but Maestral.app is also added to Login Items in System Preferences (Users & Groups prefpanel), two instances of Mastral is launched on system startup.

**To Reproduce**
Check the checkbox in GUI, add as a startup item in System Preferences. Restart computer.

**Expected behaviour**
A) Checking the GUI box would show correlate to the app being listed in Login Items.
B) App should just launch one instance and alert the user about proper behaviour

**System (please complete the following information):**
 - Maestral version: 1.2.0
 - OS: MacOS Big Sur b9

**Comment by SamSchott** [comment]
Yes, this is a known issue. There are two options to register an app to start on login on macOS: either through System Preferences or as a launch agent. The second option is more flexible because it allows us to pass command line arguments to the app before launch. Maestral allows syncing multiple Dropbox accounts by running different instances and the account to use will be passed as an argument on launch. To enable separate "start on login" entries for those we currently use launch agents. They don't show up in System Preferences but are represented as files in `~/Library/LaunchAgents`.

Having two instances of Maestral starting is not a problem. You will notice that they use the same sync process: pausing one will pause the other. Still, this can be confusing.

There are a few options to solve this:

1. Improve the documentation to clarify the "start on login" mechanism on macOS.
2. Use a System Preferences entry for the default account and launch agents for any additional accounts. This will require to different implementations on macOS and I am reluctant to maintain this. 
3. Have a single user interface to sync multiple accounts and a single "start on login" entry.

The last option is the best and it's a goal for release 2.0. However, my time is limited and this really is a hobby project...

**Comment by brgstrm** [comment]
Appreciate the reply! Agree with your conclusions 100%. 
Maestral works like a charm as-is for me, appreciate all the work that is being put into it!

**Comment by 64kramsystem** [comment]
There a similar issue on Linux.

If one configures the Systemd unit via `maestral autostart -Y`, then opens the GUI, the option `Start Maestral on login` is still unchecked, which is confusing.

Is this the same issue as above, or is it considered different (therefore, I should open a separate issue)?

**Comment by 64kramsystem** [comment]
I've noticed also another behavior (on Linux as well): if I execute `maestral gui` twice, two GUI instances are opened.

Is this again considered part of the issue above, or is it different?

**Comment by SamSchott** [comment]
> I've noticed that (on Linux) if I execute maestral gui twice, two GUI instances are opened.

That's feature, not a bug 😉 The GUI runs in a different process from the daemon and if the sync daemon is already running (for example because it has been started by another GUI instance), any new GUI instance will "attach" to the existing daemon and show a frontend for it.

You can in principle start as many GUIs as you like for a single daemon. The daemon will only shut down if you quit the GUI which started it, if any.
